pkg/resource/deploy/step_generator.go

```diff
@@ -490,9 +534,11 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, res
 	// We will also not record this non-created resource into the checkpoint as it doesn't actually
 	// exist.
 
-	if !sg.isTargetedForUpdate(urn) &&
-		!providers.IsProviderType(goal.Type) {
-
+	isTargeted := sg.isTargetedForUpdate(urn, goal.Parent, goal.Provider, goal.Dependencies)
+	if isTargeted {
+		sg.targetDependentsForUpdate(urn, goal.Parent, goal.Provider, goal.Dependencies)
+	}
```

Comment:
Let's pull this earlier (above the `hasOld` case) so we can handle the check and the bookkeeping once.